src/main/java/com/assignment/application/authenticator/VerifyUser.java

```diff
@@ -12,20 +15,29 @@
     @Autowired
     private EmployeeRepo employeeRepo;
 
-    public int authorizeUser(String username,String password){
-        if(username.equalsIgnoreCase("admin") && username.equalsIgnoreCase("admin")){
-            return 1;
-        }
-        return 0;
-    }
+    @Autowired
+    private CompanyRepo companyRepo;
 
-    public int authorizeEmployee(String username,String password){
-        Employee employee;
-        employee = employeeRepo.getEmployee(username);
-        if(employee==null || !employee.getDob().equalsIgnoreCase(password)){
+    @Autowired
+    private RedisService redisService;
+
+    public int authorizeUser(String token,String url,String type){
+        String cachedInfo = redisService.getKeyValue(StringConstant.ACCESS_TOKEN_REGEX + token);
+        if(cachedInfo.indexOf("roles: superadmin")==-1){
```

Comment:
I have a url of type {comp_id}/{emp_id}/salary. This is for posting the salary now I don't want employee to do this. For now only super admin have the power. Thats why I added the type to avoid employee from doing post operation.